### PR TITLE
Intel-MKL: add inter/intra-op and performance tuning parameters specific to MKL

### PR DESCRIPTION
**Pull request by jinghuangintel**
In this PR, We added following parameters to improve the training and evaluation out-of-box experiences on CPU and make it easier for users to control the parallelism on CPU for object detection API.
1) inter_op and intra_op to control the parallelism.
2) performance tuning parameters specific to MKL (kmp_block_time and kmp_affinity) if MKL is enabled.

### CODE REVIEW

File: research/object_detection/eval.py [hunk 1]
```diff
@@ -78,6 +78,27 @@
 flags.DEFINE_boolean('run_once', False, 'Option to only run a single pass of '
                      'evaluation. Overrides the `max_evals` parameter in the '
                      'provided config.')
+
+# Performance tuning specific to MKL
```
Comment (derekjchow) [inline research/object_detection/eval.py]: Strictly speaking, are these flags needed? It seems to me that they just set environment variables. Can the user set these variables before running the eval script?

File: research/object_detection/eval_util.py [hunk 2]
```diff
@@ -206,6 +206,8 @@ def _run_checkpoint_once(tensor_dict,
                          master='',
                          save_graph=False,
                          save_graph_dir='',
+                         inter_op=0,
```
Comment (derekjchow) [inline research/object_detection/eval_util.py]: Please update doc string with these variables.

File: research/object_detection/train.py [hunk 3]
```diff
@@ -82,6 +82,26 @@
 flags.DEFINE_string('model_config_path', '',
                     'Path to a model_pb2.DetectionModel config file.')
 
+# Performance tuning specific to MKL
```
Comment (derekjchow) [inline research/object_detection/train.py]: Same comment about eval job. Could these be set outside of the python script?

File: research/object_detection/trainer.py [hunk 4]
```diff
@@ -214,7 +214,9 @@ def train(create_tensor_dict_fn,
           worker_job_name,
           is_chief,
           train_dir,
-          graph_hook_fn=None):
+          graph_hook_fn=None,
+          inter_op=0,
```
Comment (derekjchow) [inline research/object_detection/trainer.py]: Please update doc string.

File: research/object_detection/eval.py (hunk 1 above)
Comment (jinghuangintel) [reply research/object_detection/eval.py]: @derekjchow thanks for your useful comments! yeah, those are environment variables and parallelism control parameters, very similar to https://github.com/tensorflow/benchmarks/blob/master/scripts/tf_cnn_benchmarks/benchmark_cnn.py (line 178, 323),  we are trying to improve user's OOB experiences while they use MKL.

File: research/object_detection/eval_util.py (hunk 2 above)
Comment (jinghuangintel) [reply research/object_detection/eval_util.py]: thanks for the comments, i have added the inter_op/intra_op in the doc string already.

**Comment by jinghuangintel** [comment]
@derekjchow thanks for your comments. i have also squashed multiple PR into one single PR already, let me know if you have any further suggestions. thanks.

**Pull request closed without merging**